.buildkite/pipeline.test.yml

```diff
@@ -1,7 +1,8 @@
-- label: ":book: LinkCheck"
-  instance_size: small
-  commands:
-    - export LINT=1
-    - ./ci/env/install-dependencies.sh
-    - ./ci/ci.sh check_sphinx_links
-  soft_fail: True
+name: ":book: LinkCheck"
+description: "Check the integrity of all external doc links with sphinx"
+steps:
+  - label: "Run linkcheck"
```

Comment:
can we move this to lint.rayci.yaml or build.rayci.yml, the pipeline.test.yml is on CIv1 and doesn't have the job_env 

also I don't think the `description` field is supported

our CI only supports a selected set of buildkite field due to security reason